Feature/pt1/project setup

flutterVlad opened the pull request:
Video:
https://github.com/flutterVlad/food_app/assets/137285738/2f311a49-c8dd-4429-ace3-79334004fb75

flutterVlad committed:
[PT1] feature/PT1/project-setup

- Setup project architecture
- Connect Firebase
- Create home screen UI

flutterVlad committed:
[PT1] feature/PT1/project-setup

- Setup project architecture
- Connect Firebase
- Create home screen UI

+

- Add Firebase config in DI

AliaksandrHuzich commented on core/lib/core.dart:
```diff
@@ -1,7 +1,4 @@
 library core;
```
formatting

AliaksandrHuzich commented on core/lib/di/data_di.dart:
```diff
@@ -0,0 +1,13 @@
+import 'package:get_it/get_it.dart';
```
formatting

AliaksandrHuzich commented on core/lib/di/data_di.dart:
```diff
@@ -0,0 +1,13 @@
+import 'package:firebase_core/firebase_core.dart';
+import 'package:core/config/firebase_options.dart';
+
+final FirebaseDI firebaseDI = FirebaseDI();
```
datadi

AliaksandrHuzich commented on core/lib/di/data_di.dart:
```diff
@@ -0,0 +1,13 @@
+import 'package:core/config/firebase_options.dart';
+
+final FirebaseDI firebaseDI = FirebaseDI();
+
```
formatting

AliaksandrHuzich commented on core_ui/lib/src/theme/app_theme.dart:
```diff
@@ -0,0 +1,26 @@
+      ),
+      primarySwatch: Colors.orange,
+      bottomNavigationBarTheme: BottomNavigationBarThemeData(
+        backgroundColor: Colors.black54,
```
colors to separate file

AliaksandrHuzich commented on core_ui/lib/src/widgets/home_card_element.dart:
```diff
@@ -0,0 +1,71 @@
+import 'package:flutter/material.dart';
```
from core ui to widgets folder with screen

AliaksandrHuzich commented on core_ui/lib/src/widgets/home_card_element.dart:
```diff
@@ -0,0 +1,71 @@
+  Widget build(BuildContext context) {
+    return Container(
+      padding: const EdgeInsets.all(12),
+      width: 170,
```
fixed width

AliaksandrHuzich commented on core_ui/lib/src/widgets/widgets.dart:
```diff
@@ -0,0 +1 @@
+export 'home_card_element.dart';
```
formatting

AliaksandrHuzich commented on data/lib/data.dart:
```diff
@@ -1,7 +1,3 @@
 library data;
```
formatting

AliaksandrHuzich commented on domain/lib/domain.dart:
```diff
@@ -1,7 +1,3 @@
 library domain;
```
formatting

AliaksandrHuzich commented on domain/lib/domain.dart:
```diff
@@ -1,7 +1,3 @@
 library domain;
 
-/// A Calculator.
```
formatting

AliaksandrHuzich commented on features/home/lib/src/bloc/home/bloc.dart:
```diff
@@ -0,0 +1,19 @@
+import 'package:bloc/bloc.dart';
```
rewrite this

AliaksandrHuzich commented on features/home/lib/src/bloc/home/event.dart:
```diff
@@ -0,0 +1,3 @@
+abstract class HomeEvent {}
```
formatting

AliaksandrHuzich commented on features/home/lib/src/bloc/home/state.dart:
```diff
@@ -0,0 +1,9 @@
+class HomeState {
```
rewrite this

AliaksandrHuzich commented on features/home/lib/src/bloc/home/view.dart:
```diff
@@ -0,0 +1,23 @@
+import 'event.dart';
+import 'state.dart';
+
+class HomePage extends StatelessWidget {
```
2 widgets (screen & content)

AliaksandrHuzich commented on features/home/lib/src/bloc/home/view.dart:
```diff
@@ -0,0 +1,23 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+import 'bloc.dart';
```
imports

flutterVlad committed:
[PT1] feature/PT1/project-setup

- Setup project architecture
- Connect Firebase
- Create home screen UI

+

- Add Firebase config in DI

+

- Add functionality for downloading data from Firebase
